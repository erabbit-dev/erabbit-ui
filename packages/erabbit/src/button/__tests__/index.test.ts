import { Button } from '..'
import { mount } from '@vue/test-utils'

test('should render default button', () => {
  const wrapper = mount(Button, {
    slots: {
      default: 'Submit',
    },
  })
  expect(wrapper.html()).toMatchSnapshot()
})

test('should trigger click event', () => {
  const wrapper = mount(Button, {
    slots: {
      default: 'Submit',
    },
    props: {
      size: 'large',
      type: 'primary',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
})
