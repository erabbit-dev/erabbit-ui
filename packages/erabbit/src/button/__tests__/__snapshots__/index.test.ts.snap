// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render default button 1`] = `
"<button class=\\"er-button ellipsis er-button--middle er-button--default\\">
  <!---->Submit
</button>"
`;

exports[`should trigger click event 1`] = `
"<button class=\\"er-button ellipsis er-button--large er-button--primary\\">
  <!---->Submit
</button>"
`;
